src/components/NavigationStepper/index.tsx

```diff
@@ -0,0 +1,223 @@
+Step.displayName = 'NavigationStepper.Step'
+
+Step.propTypes = {
+  children: PropTypes.node.isRequired,
+  isLoading: PropTypes.bool,
+}
+
+const StyledUl = styled('ul', {
+  shouldForwardProp: prop => !['count', 'condensed'].includes(prop.toString()),
+})<{ count: number; condensed?: boolean }>`
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  width: 100%;
+
+  ${({ count, condensed, theme }) =>
+    !condensed
+      ? css`
+          display: grid;
+          grid-template-columns: repeat(${count}, 1fr);
+          ${StyledStep} {
+            place-self: center;
+            &:first-child {
+              place-self: start;
+            }
+            &:last-child {
+              place-self: end;
+            }
+          }
+        `
+      : css`
+          display: flex;
+          ${StyledStep} {
+            margin-right: ${theme.space[1]};
+            &:last-child {
+              margin-right: 0;
+            }
+          }
+        `}
+`
+
+type NavigationStepperProps = {
+  step?: number
+  children: ReactNodeArray
+  condensed?: boolean
+}
+
+type NavigationStepperType = FunctionComponent<NavigationStepperProps> & {
+  Step: VoidFunctionComponent<StepProps>
+}
+
+const NavigationStepper: NavigationStepperType = ({
+  condensed,
+  children,
+  step = 1,
+}) => {
+  const theme = useTheme()
+  const isMobile = useMedia(
+    [`(max-width: ${theme.screens.small}px)`],
+    [true],
+    false,
+  )
+
+  console.log('IsMobile', isMobile)
+  const computedCondensed = condensed !== undefined ? condensed : isMobile
+  const steps = React.Children.map<JSX.Element | null, ReactNode>(
+    children,
+    (child, index) => {
+      if (isValidElement(child) && child.type === Step) {
+        const childStep = index + 1
+        const isCompleted = step > childStep
+        const isCurrent = step === childStep
+        const { isLoading = false } = child.props as StepProps
+
+        return (
+          <StyledStep
+            isLoading={isLoading}
+            isCompleted={isCompleted}
+            aria-current={isCurrent}
+            condensed={computedCondensed}
+          >
+            <StyledStepNumber>
+              {isLoading && isCurrent ? (
+                <StyledStepLoader>
+                  <ActivityIndicator
+                    active
+                    trailColor="transparent"
+                    color="success"
+                    size="100%"
+                    strokeWidth={8}
+                  />
+                </StyledStepLoader>
+              ) : null}
+              {isCompleted ? (
+                <Icon stroke="currentColor" strokeWidth={2} name="check" />
+              ) : (
+                childStep
+              )}
+            </StyledStepNumber>
+            <StyledStepContent aria-hidden={!isCurrent && computedCondensed}>
+              {child}
+            </StyledStepContent>
+          </StyledStep>
+        )
+      }
+
+      return null
+    },
+  ) as ReactNode[]
```

Comment:
If you filter out null values would you still need to cast as ReactNode[] ?

Reply:
No it is `steps.length` the problem cause React.Children.map can return undefined

Reply:
Alright :)